examples/platforms/kw41z/radio.c

```diff
@@ -234,24 +234,25 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
 
     sState = OT_RADIO_STATE_RECEIVE;
 
-    otEXPECT(rf_get_state() != XCVR_RX_c);
-
-    rf_abort();
-
-    /* Set Power level for auto TX */
-    rf_set_tx_power(sAutoTxPwrLevel);
-    rf_set_channel(aChannel);
-    sRxFrame.mChannel = aChannel;
+    /* Check if the channel needs to be changed */
+    if (sChannel != aChannel)
```

Comment:
We set **sState = OT_RADIO_STATE_RECEIVE** right above this **if** statement. @jwhui had originally suggested that I move the calls to set the registers within this **if** statement, but I wasn't sure whether or not I should include

**ZLL->RX_FRAME_FILTER &= ~(ZLL_RX_FRAME_FILTER_ACK_FT_MASK);**
**ZLL->IRQSTS = ZLL->IRQSTS;**
**ZLL->PHY_CTRL |= XCVR_RX_c;**
**ZLL->PHY_CTRL &= ~ZLL_PHY_CTRL_SEQMSK_MASK;**

within the **if** statement. I interpreted those calls as necessary to prepare the physical layer for receiving, so I wondered whether they should be set regardless. I can't tell from your two possible suggested changes whether setting **sState = OT_RADIO_STATE_RECEIVE** before this **if** statement is sufficient to enable receiving, or if we need the aforementioned register changes unconditionally as well. In fact, one of your suggestions for **...|| (rf_get_state() != XCVR_RX_c)** implies the latter.

My proposed solution would be to move those four lines below the **if** statement, which I expect achieves the same desired outcome as your suggestion; this however seems to contradict what Jonathan had recommended regarding only setting the registers if the channel differed. I have tested this proposed solution, and it commissions and communicates on Channel 11 from boot without a problem (your point of concern). Please let me know what you think, and thank you for your time and review.

Reply:
@tfrazel, I think you interpreted my comment too literally :)

If the radio is not in the RX sequence, then I think it is fine to write out the registers again.  In this case, I consider the channel to be "undefined" and, as a result, the desired channel differs from the current receive channel.